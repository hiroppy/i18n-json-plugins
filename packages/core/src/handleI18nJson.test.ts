import { tmpdir } from "node:os";
import { mkdtemp, rmdir } from "node:fs/promises";
import { join } from "node:path";
import {
  createJson,
  addKey,
  getContent,
  removeEmptyKey,
} from "./handleI18nJson";

describe("handleI18nJson", () => {
  let tmpPath = "";

  beforeEach(async () => {
    tmpPath = await mkdtemp(join(tmpdir(), "i18n-json-plugins"));
  });

  afterEach(async () => {
    await rmdir(tmpPath, { recursive: true });
  });

  test("should create a JSON file", async () => {
    await createJson(tmpPath, "test");

    expect(await getContent(tmpPath, "test")).toMatchSnapshot();
  });

  test("should add keys to a nested object", async () => {
    await createJson(
      tmpPath,
      "test",
      JSON.stringify({
        components: {},
      })
    );
    await addKey(tmpPath, "test", "foo");
    await addKey(tmpPath, "test", "foo.bar");
    await addKey(tmpPath, "test", "foo.bar.baz");
    await addKey(tmpPath, "test", "components.foo");

    expect(await getContent(tmpPath, "test")).toMatchSnapshot();
  });

  test("should throw an error when a parent has already had a value", async () => {
    await createJson(
      tmpPath,
      "test",
      JSON.stringify({
        components: "components",
      })
    );
    await expect(
      addKey(tmpPath, "test", "components.foo")
    ).rejects.toThrowError("components(parent) has already had a value");
  });

  test("should throw an error when a parent doesn't exist", async () => {
    await createJson(tmpPath, "test", JSON.stringify({}));
    await expect(
      addKey(tmpPath, "test", "components.foo")
    ).rejects.toThrowError('components(parent) doesn\'t exist on "test"');
  });

  test("should remove empty keys from an object", () => {
    expect(
      removeEmptyKey({
        foo: "",
        bar: {},
        nested: {
          foo: "",
          bar: {
            baz: "",
          },
        },
      })
    ).toMatchSnapshot();
  });
});
