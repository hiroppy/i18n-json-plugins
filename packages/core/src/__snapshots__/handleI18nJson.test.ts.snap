// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleI18nJson should add keys to a nested object 1`] = `
"{
  \\"components\\": {
    \\"foo\\": \\"\\"
  },
  \\"foo\\": {
    \\"bar\\": {
      \\"baz\\": \\"\\"
    }
  }
}"
`;

exports[`handleI18nJson should create a JSON file 1`] = `"{}"`;

exports[`handleI18nJson should remove empty keys from an object 1`] = `
Object {
  "bar": Object {},
  "nested": Object {
    "bar": Object {},
  },
}
`;
